core/templates/dev/head/domain/exploration/QuestionListObjectFactory.js

```diff
@@ -35,6 +35,14 @@ oppia.factory('QuestionListObjectFactory', [function() {
     this._questions.push(question);
   };
 
+  QuestionList.prototype.removeQuestion = function(question){
+    var index = this._questions.indexOf(question);
+    if(index != -1){
+      return this._questions.splice(index,1);
+    }
+    return ;
```

Comment:
Yes, that would be better. Should I use 'try-catch' wherever I am using this method?